File: Part1_강의자료/Ch08_투포인터/문제별코드/2003_수들의합2/2003.java
import java.util.Scanner;

class Main
{
    public static void main (String[] args) {
        Scanner sc = new Scanner(System.in);

        int N = sc.nextInt();
        int M = sc.nextInt();
        int[] arr = new int[N];
        for (int i = 0; i < N; i++)
            arr[i] = sc.nextInt();

        int ans = 0;
        int nextIndex = 0;
        int currentSum = 0;
        for (int i = 0; i < N; i++) {
            while (currentSum < M && nextIndex < N)
                currentSum += arr[nextIndex++];
            if (currentSum == M) ans++;
            currentSum -= arr[i];
        }

        System.out.println(ans);
    }
}

File: Part1_강의자료/Ch08_투포인터/문제별코드/2003_수들의합2/2003.cpp
#include <stdio.h>

int main()
{
    int N, M, arr[10005];
    scanf("%d %d", &N, &M);
    for (int i = 0; i < N; i++)
        scanf("%d", &arr[i]);

    int next_index = 0, current_sum = 0, ans = 0;
    for (int i = 0; i < N; i++)
    {
        while (current_sum < M && next_index < N)
            current_sum += arr[next_index++];
        if (current_sum == M)
            ans++;
        current_sum -= arr[i];
    }

    printf("%d\n", ans);
    return 0;
}
